### ISSUE TITLE
Annotation tooltip position

### ISSUE THREAD

**Issue by markov00** [elastic/elastic-charts#297]
**Is your feature request related to a problem? Please describe.**
The current implementation place the annotation tooltip, to the left or to the right of the annotation area (if it's a rectangle) or place it just in the middle for a line annotation.
With that approach, the tooltip can sometime overlap with the current tooltip, or be shown in some weird places

<img width="512" alt="Screenshot 2019-08-06 at 18 33 38" src="https://user-images.githubusercontent.com/1421091/62560023-7632b180-b87c-11e9-9aba-37f6fd0511a1.png">

<img width="423" alt="Screenshot 2019-08-06 at 18 51 03" src="https://user-images.githubusercontent.com/1421091/62560020-759a1b00-b87c-11e9-8de4-9c0b057107e3.png">
<img width="466" alt="Screenshot 2019-08-06 at 18 35 02" src="https://user-images.githubusercontent.com/1421091/62560021-759a1b00-b87c-11e9-962e-401578b6a848.png">

**Describe the solution you'd like**
I'd like to check if we can move that tooltip to a proper place. 

**Describe alternatives you've considered**
Possible solutions:
- if a data tooltip is already shown, move the tooltip text inside that tooltip (on the header, on on a different row)
- if no data tooltip is shown, display the tooltip but not stocked with the annotation edge. maybe it's better to have it moving together with the mouse.
- avoid showing a real tooltip, and displaying the text somewhere else on the chart: on the title? just below, above the axis?

**Additional context**
The vislib chart incorporate that annotation inside the tooltip header:
<img width="455" alt="Screenshot 2019-08-06 at 19 04 33" src="https://user-images.githubusercontent.com/1421091/62560321-14267c00-b87d-11e9-9cfd-6da3df866cee.png">
<img width="452" alt="Screenshot 2019-08-06 at 19 04 29" src="https://user-images.githubusercontent.com/1421091/62560322-14267c00-b87d-11e9-92ec-02387d399f27.png">

**Kibana Cross Issues**
n/a

**Checklist**

- [x] this request is checked against already exist requests
- [x] every related Kibana issue is listed under `Kibana Cross Issues` list
- ~[ ] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present`

**Comment by markov00** [comment]
cc @cchaos

**Comment by cchaos** [comment]
I think that the chart itself should take precedence. Only show the annotation's tooltip if the cursor is not also hovering over part of the chart.

**Comment by markov00** [comment]
@nickofthyme I'm resurrecting this enhancement: now that we have the popperjs is that something that can be handled by that library?

**Comment by nickofthyme** [comment]
Yes, you _could_ change the anchor of the annotations tooltip. But I think this looks hideous and is prone for very wide tooltips. I think this should be handled in the tooltip redesign, to incorporate the annotation into the tooltip like visualize currently does. But do it in a better, more customizable way.
